drivers/spi/spi_nrfx_spim.c

```diff
@@ -568,21 +575,20 @@ static const struct spi_driver_api spi_nrfx_driver_api = {
 	.release = spi_nrfx_release,
 };
 
-#ifdef CONFIG_PM_DEVICE
 static int spim_nrfx_pm_action(const struct device *dev,
 			       enum pm_device_action action)
 {
-	int ret = 0;
+	int ret = -ENOTSUP;
 	struct spi_nrfx_data *dev_data = dev->data;
 	const struct spi_nrfx_config *dev_config = dev->config;
 
 	switch (action) {
+	case PM_DEVICE_ACTION_TURN_ON:
```

Comment:
it's not needed any more. Function can return `-ENOTSUP` in that case and it's accepted by the PM.

Reply:
aligned to scheme used in PWM and UARTE drivers